core/src/main/scala/kafka/log/Log.scala

```diff
@@ -139,7 +140,13 @@ class Log(val dir: File,
   /* Load the log segments from the log files on disk */
   private def loadSegments() {
     // create the log directory if it doesn't exist
-    dir.mkdirs()
+    if (!Files.exists(dir.toPath())) {
+      try {
+          Files.createDirectory(dir.toPath())    
```

Comment:
Ah okay. I didn't look into the difference between mkdirs/mkdir and hence the confusion. I will change it to createDirectories to be consistent with the original intent.